lightning/src/ln/channelmanager.rs

```diff
@@ -1487,6 +1487,28 @@ impl<'a> ChannelManager<'a> {
 		events.append(&mut new_events);
 	}
 
+	/// Latency/peer disconnection may trigger us to mark as disabled some
+	/// of our channels. After some time, if channels are still disabled
+	/// we need to broadcast ChannelUpdate to inform other network peers
+	/// about the uselessness of this channels.
+	pub fn timer_chan_freshness_every_min(&self) {
+		let _ = self.total_consistency_lock.read().unwrap();
+		{
+			let mut channel_state_lock = self.channel_state.lock().unwrap();
+			let channel_state = channel_state_lock.borrow_parts();
+			for (_, chan) in channel_state.by_id {
+				// Check if channel is still disabled
+				if !chan.is_live() {
```

Comment:
This may have happened one second ago, though? The channel needs to have been disabled at the last call to this fn.

Reply:
Right so you need to track pending_disabled_chan in ChannelManager between the calls but it may imply we may worst-case take up to ~120 seconds to announce a ChannelUpdate?

Reply:
Yep. I think thats totally fine, but if you are worried about it you can also halve the time and make the user call every 30 seconds and then its worst-case 1.5 minutes (but I dont think it matters).